kroxylicious-kms-provider-kroxylicious-inmemory/src/main/java/io/kroxylicious/kms/provider/kroxylicious/inmemory/UuidKekId.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.kroxylicious.inmemory;
+
+import java.util.Objects;
+import java.util.UUID;
+
+import io.kroxylicious.kms.service.KekId;
+import io.kroxylicious.kms.service.KmsException;
+
+public record UuidKekId(UUID keyId) implements KekId {
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <K> K getId(Class<K> keyType) {
+        if (!keyType.isAssignableFrom(UUID.class)) {
+            throw new KmsException("Unsupported keyType (" + keyType + ") requested. Only UUID's are supported");
+        }
+        return (K) this.keyId;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Why override?

Reply:
Removed in 5a11ea4e88cf4219528390d5b32788a5a1c34807